src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs

```diff
@@ -319,9 +325,12 @@ public async Task TestIncrementalUpdateHandlesReferenceReversal()
                 await remoteWorkspace.CurrentSolution.CompilationState.GetChecksumAsync(CancellationToken.None));
 
             // reverse project references again and incrementally update
-            solution = solution.RemoveProjectReference(projectId2, project2ToProject1);
-            solution = solution.AddProjectReference(projectId1, project1ToProject2);
-            await workspace.ChangeSolutionAsync(solution);
+            workspace.SetCurrentSolution(solution =>
+            {
+                solution = solution.RemoveProjectReference(projectId2, new ProjectReference(projectId1));
+                return solution.AddProjectReference(projectId1, new ProjectReference(projectId2));
+            }, WorkspaceChangeKind.SolutionChanged);
```

Comment:
not quite getting why these tests changed?

Reply:
The reason is that SetCurrentSolutionEx does not use the transformation, which post-processes the new solution snapshot (e.g. handles linked files and now also initialized fallback options). Hence, the test ended up with snapshots that did not have fallback options initialized. These then differ in checksum.

As I noted in the PR description one follow up I plan to do is to remove intenral usage of SetCurrentSolutionEx (which means updating quite a few tests).

Reply:
gotcha.  i def approve of removing that method.